ietf/secr/sreq/forms.py

```diff
@@ -109,31 +93,44 @@ def __init__(self, group, *args, **kwargs):
         group_acronym_choices = [('','--Select WG(s)')] + other_groups
-        for i in range(1, 5):
-            self.fields['wg_selector{}'.format(i)].choices = group_acronym_choices
-
-        # disabling handleconflictfield (which only enables or disables form elements) while we're hacking the meaning of the three constraints currently in use:
-        #self.fields['wg_selector1'].widget.attrs['onChange'] = "document.form_post.conflict1.value=document.form_post.conflict1.value + ' ' + this.options[this.selectedIndex].value; return handleconflictfield(1);"
-        #self.fields['wg_selector2'].widget.attrs['onChange'] = "document.form_post.conflict2.value=document.form_post.conflict2.value + ' ' + this.options[this.selectedIndex].value; return handleconflictfield(2);"
-        #self.fields['wg_selector3'].widget.attrs['onChange'] = "document.form_post.conflict3.value=document.form_post.conflict3.value + ' ' + this.options[this.selectedIndex].value; return handleconflictfield(3);"
-        self.fields['wg_selector1'].widget.attrs['onChange'] = "document.form_post.conflict1.value=document.form_post.conflict1.value + ' ' + this.options[this.selectedIndex].value; return 1;"
-        self.fields['wg_selector2'].widget.attrs['onChange'] = "document.form_post.conflict2.value=document.form_post.conflict2.value + ' ' + this.options[this.selectedIndex].value; return 1;"
-        self.fields['wg_selector3'].widget.attrs['onChange'] = "document.form_post.conflict3.value=document.form_post.conflict3.value + ' ' + this.options[this.selectedIndex].value; return 1;"
-        self.fields['wg_selector4'].widget.attrs['onChange'] = "document.form_post.joint_with_groups.value=document.form_post.joint_with_groups.value + ' ' + this.options[this.selectedIndex].value; return 1;"
-
-        # disabling check_prior_conflict javascript while we're hacking the meaning of the three constraints currently in use
-        #self.fields['wg_selector2'].widget.attrs['onClick'] = "return check_prior_conflict(2);"
-        #self.fields['wg_selector3'].widget.attrs['onClick'] = "return check_prior_conflict(3);"
-        
+        self.fields['joint_with_groups_selector'].choices = group_acronym_choices
+
+        # Set up constraints for the meeting
+        self._wg_field_data = []
+        for constraintname in meeting.session_constraintnames.all():
+            # two fields for each constraint: a CharField for the group list and a selector to add entries
+            constraint_field = forms.CharField(max_length=255, required=False)
+            # add a new class, taking care in case some already exist
+            existing_classes = constraint_field.widget.attrs.get('class', '').split()
+            constraint_field.widget.attrs['class'] = ' '.join(existing_classes + ['wg_constraint'])
+            constraint_field.widget.attrs['data-slug'] = constraintname.slug
+            constraint_field.widget.attrs['data-constraint-name'] = str(constraintname).title()
```

Comment:
Makes sense. I'm not sure how much you had in mind. I extracted out an `_add_widget_class()` because that seemed like a well-defined operation. The other attr setting seems straightforward as it is and "add_class_and_set_slug" seemed oddly specific.

Reply:
I had been thinking of extracting more extensive, but I'll accept that you're right in this situation.